src/NuGet.Services.EndToEnd/Support/CommandRunner.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit.Abstractions;
+
+namespace NuGet.Services.EndToEnd.Support
+{
+    public static class CommandRunner
+    {
+        public static async Task<CommandRunnerResult> RunAsync(
+            string fileName,
+            string workingDirectory,
+            string arguments,
+            ITestOutputHelper logger,
+            IDictionary<string, string> environmentVariables = null,
+            int timeOutInMilliseconds = 60000)
+        {
+            await Task.Yield();
```

Comment:
To ensure that the whole process execution happens on a different thread than the caller. Otherwise, all the code up until the first `await` will happen on the calling thread.